src/Umbraco.Web.UI.Client/src/common/directives/components/forms/umbfocuslock.directive.js

```diff
@@ -16,53 +17,212 @@
+
+                // Options for the observer (which mutations to observe)
+                var config = { attributes: true, childList: true, subtree: true};
+
+                // Whenever the DOM changes ensure the list of focused elements is updated
+                function domChange() {
+                    getFocusableElements();
+                }
+
+                // Start observing the target node for configured mutations
+                observer.observe(target, config);
+
+                // Disconnect observer
+                if(disconnectObserver){
+                    observer.disconnect();
+                }
+            }
+
+            function cleanupEventHandlers() {
+                var activeEditor = infiniteEditors[infiniteEditors.length - 1];
+                var inactiveEditors = infiniteEditors.filter(editor => editor !== activeEditor);
+
+                if(inactiveEditors.length > 0) {
+                    for (var index = 0; index < inactiveEditors.length; index++) {
+                        var inactiveEditor = inactiveEditors[index];
+
+                        // Remove event handlers from inactive editors
+                        inactiveEditor.removeEventListener('keydown', handleKeydown);
+                    }
+                }
+                else {
+                    // Remove event handlers from the active editor
+                    activeEditor.removeEventListener('keydown', handleKeydown);
+                }
+            }
+
+            function onInit(targetElm) {
+
+                $timeout(() => {
+
+                    // Fetch the DOM nodes we need
+                    getDomNodes();
+
+                    // Cleanup event handlers if we're in infinite editing mode
+                    if(infiniteEditors.length > 0){
+                        cleanupEventHandlers();
+                    }
+
+                    getFocusableElements(targetElm);
+
+                    if(focusableElements.length > 0) {
+
+                        observeDomChanges();
+
+                        setElementFocus();
+
+                        //  Handle keydown
+                        target.addEventListener('keydown', handleKeydown);
+                    }
+
+                });
+            }
+
+            scope.$on('$includeContentLoaded', () => {
+                angularHelper.safeApply(scope, () => {
+                    onInit();
+                });
+            });
+
+            // If more than one editor is still open then re-initialize otherwise remove the event listener
+            scope.$on('$destroy', function () {
+                // Make sure to disconnect the observer so we potentially don't end up with having many active ones
+                disconnectObserver = true;
+
+                // Pass the correct editor in order to find the focusable elements
+                var newTarget = infiniteEditors[infiniteEditors.length - 2];
```

Comment:
@nul800sebastiaan I've just experienced a problem with this line.
From reading this line there is an obvious mistake as we easily could have just one or zero infinite editors.

Lets me know if you need anything from me, but I think this will have to be dealt with before making the RC. :-)